lib/services/key_recovery_service.ts

```diff
@@ -370,12 +418,17 @@ export class SNKeyRecoveryService extends PureService {
       KeyRecoveryStrings.KeyRecoveryLoginFlowPrompt(keyParams),
       KeyRecoveryStrings.KeyRecoveryPasswordRequired
     );
+
     const response = await this.challengeService
       .promptForChallengeResponse(challenge);
     if (!response) {
-      return { success: false };
+      resolve({ success: false });
     }
-    const password = response.values[0].value as string;
+
+    /** Dismiss challenge */
+    this.challengeService.cancelChallenge(challenge);
```

Comment:
See my previous comment about cancelling custom challenges that are actually completed